profile-diff/script.js

```diff
@@ -0,0 +1,320 @@
+      width: '71px',
+    },
+  });
+  CONTAINER.appendChild(displayImage);
+
+  const username = createElement({
+    type: 'div',
+    attributes: {
+      class: 'user_name',
+    },
+  });
+  username.innerHTML = `${user.first_name} ${user.last_name}`;
+  CONTAINER.appendChild(username);
+
+  const expandControlContainer = createElement({
+    type: 'div',
+    attributes: { class: 'expand-control-container' },
+  });
+  const expandTextDiv = createElement({
+    type: 'div',
+    attributes: { class: 'expand-control-text' },
+  });
+  expandControlContainer.appendChild(expandTextDiv);
+  expandTextDiv.innerHTML = `See ${
+    expanded ? 'Less' : 'All'
+  }<i class="fa fa-chevron-${
+    expanded ? 'up' : 'down'
+  } expand-control-icon"></i>`;
+
+  expandTextDiv.addEventListener('click', () => {
+    expanded = !expanded;
+    expandTextDiv.innerHTML = `See ${
+      expanded ? 'Less' : 'All'
+    }<i class="fa fa-chevron-${
+      expanded ? 'up' : 'down'
+    } expand-control-icon"></i>`;
+    renderProfileDetails(user, profileDiff);
+  });
+  CONTAINER.appendChild(expandControlContainer);
+  renderProfileDetails(user, profileDiff);
+  reasonHeading = createElement({
+    type: 'p',
+    attributes: {
+      class: 'reason_heading',
+    },
+  });
+  reasonHeading.innerHTML = 'Reason ( Optional )';
+  CONTAINER.appendChild(reasonHeading);
+
+  const reasonTextArea = createElement({
+    type: 'textarea',
+    attributes: {
+      class: 'reason_textarea',
+      rows: 4,
+    },
+  });
+  CONTAINER.appendChild(reasonTextArea);
+
+  const buttonsDiv = createElement({
+    type: 'div',
+    attributes: {
+      class: 'button__container',
+    },
+  });
+
+  const approvalButton = createElement({
+    type: 'button',
+    attributes: {
+      class: 'button__approve',
+    },
+  });
+  approvalButton.innerHTML = 'Approve';
+  buttonsDiv.appendChild(approvalButton);
+
+  const rejectButton = createElement({
+    type: 'button',
+    attributes: {
+      class: 'button__reject',
+    },
+  });
+  rejectButton.innerHTML = 'Reject';
+  buttonsDiv.appendChild(rejectButton);
+
+  approvalButton.onclick = async () => {
+    approvalButton.innerHTML = '<div class="button__loader"></div>';
+    approvalButton.disabled = true;
+    rejectButton.disabled = true;
+    try {
+      const response = await fetch(`${API_BASE_URL}/users/${user.id}`, {
+        method: 'PATCH',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        credentials: 'include',
+        body: JSON.stringify({
+          id: profileDiff.id,
+          message: reasonTextArea.value,
+        }),
+      });
+
+      if (response.ok) {
+        showToast({
+          type: 'success',
+          message: 'The Profile changes are Approved!',
+        });
+        setTimeout(() => {
+          window.location.href = '/profile-diffs';
+        }, 3000);
+      } else {
+        showToast({ type: 'error', message: 'Something went wrong!' });
```

Comment:
- can we move this api call into a common file where all api calls are handled? makes it easy to reuse and reduces visual clutter.

Reply:
I moved the API calls for specific pages into a separate file. We can create a different PR for refactoring this as all other pages are also making the API calls from local utils, we have to refactor all of that and this will increase the scope of my PR.